networkx/algorithms/connectivity/kcutsets.py

```diff
@@ -0,0 +1,221 @@
+    Once we find a minimum cut, we add an edge between the high degree
+    node and the target node of the local maximum flow computation to make 
+    sure that we will not find that minimum cut again.
+
+    See also
+    --------
+    node_connectivity
+    edmonds_karp
+    shortest_augmenting_path
+
+    References
+    ----------
+    .. [1]  Kanevsky, A. (1993). Finding all minimum-size separating vertex 
+            sets in a graph. Networks 23(6), 533--541.
+            http://onlinelibrary.wiley.com/doi/10.1002/net.3230230604/abstract
+
+    """
+    if not nx.is_connected(G):
+        raise nx.NetworkXError('Input graph is disconnected.')
+
+    # Initialize data structures.
+    # Keep track of the cuts already computed so we do not repeat them.
+    seen = []
+    # Even-Tarjan reduction is what we call auxiliary digraph 
+    # for node connectivity.
+    H = build_auxiliary_node_connectivity(G)
+    mapping = H.graph['mapping']
+    R = build_residual_network(H, 'capacity')
+    kwargs = dict(capacity='capacity', residual=R)
+    # Define default flow function
+    if flow_func is None:
+        flow_func = default_flow_func
+    if flow_func is shortest_augmenting_path:
+        kwargs['two_phase'] = True
+    # Begin the actual algorithm
+    # step 1: Find node connectivity k of G
+    if k is None:
+        k = nx.node_connectivity(G, flow_func=flow_func)
+    # step 2: 
+    # Find k nodes with top degree, call it X:
+    degree = G.degree().items()
+    X = set(n for n, d in sorted(degree, key=itemgetter(1), reverse=True)[:k])
+    # Check if X is a k-node-cutset
+    if _is_separating_set(G, X):
+        seen.append(X)
+        yield X
+
+    for x in X:
+        # step 3: Compute local connectivity flow of x with all other
+        # non adjacent nodes in G
+        non_adjacent = set(G) - X - set(G[x])
+        for v in non_adjacent:
+            # step 4: compute maximum flow in an Even-Tarjan reduction H of G
+            # and step:5 build the associated residual network R
+            R = flow_func(H, '%sB' % mapping[x], '%sA' % mapping[v], **kwargs)
+            flow_value = R.graph['flow_value']
+
+            if flow_value == k:
+                ## Remove saturated edges form the residual network
+                saturated_edges = [(u, w, d) for (u, w, d) in
+                                    R.edges_iter(data=True)
+                                    if d['capacity'] == d['flow']]
+                R.remove_edges_from(saturated_edges)
+                # step 6: shrink the strongly connected components of 
+                # residual flow network R and call it L
+                L = nx.condensation(R)
+                cmap = L.graph['mapping']
+                # step 7: Compute antichains of L; they map to closed sets in H
+                # Any edge in H that links a closed set is part of a cutset
+                for antichain in antichain_generator(L):
```

Comment:
we can change antichain_generator(L) to nx.antichains(L) and remove antichain_generator and transitive_closure from this file